Source/WebCore/platform/graphics/TrackPrivateBase.h

```diff
@@ -80,9 +85,34 @@ class WEBCORE_EXPORT TrackPrivateBase
     WTFLogChannel& logChannel() const final;
 #endif
 
+    using Task = Function<void(TrackPrivateBaseClient&)>;
+    void notifyClients(Task&&);
+    void notifyMainThreadClient(Task&&);
+
 protected:
     TrackPrivateBase() = default;
 
+    template <typename T>
+    class Shared final : public ThreadSafeRefCounted<Shared<T>> {
```

Comment:
Did you just re-invent WTF::Box? At least it looks super similar so it's worth double checking.